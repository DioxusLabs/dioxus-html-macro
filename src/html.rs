use crate::prelude::*;

#[derive(Default)]
pub struct Html {
    pub elements: Vec<Item>,
}

impl Parse for Html {
    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
        let mut elements = vec![];
        while !input.is_empty() && !input.peek2(Token![/]) {
            elements.push(input.parse()?);
        }
        Ok(Html { elements })
    }
}

impl ToTokens for Html {
    fn to_tokens(&self, tokens: &mut TokenStream) {
        for element in &self.elements {
            tokens.extend(quote! {
                #element
            });
        }
    }
}
