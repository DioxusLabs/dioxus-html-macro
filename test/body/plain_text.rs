use dioxus_html_macro::html;

fn main() {
    html!(
        <div>
           oops, I forgot to wrap it in quotes
        </div>
    );
}
