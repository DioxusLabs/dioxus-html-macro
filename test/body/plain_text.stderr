error: expected string literal, found identifier. hint: try wrapping the text in quotes.
 --> test/body/plain_text.rs:6:12
  |
6 |            oops, I forgot to wrap it in quotes
  |            ^^^^
